src/main/java/li/cil/sedna/riscv/R5CPUTemplate.java

```diff
@@ -402,6 +419,12 @@ private void interpretTrace32(final MemoryMappedDevice device, int inst, long pc
     private void interpretTrace64(final MemoryMappedDevice device, int inst, long pc, int instOffset, final int instEnd, final LongSet breakpoints) {
         try { // Catch any exceptions to patch PC field.
             for (; ; ) { // End of page check at the bottom since we enter with a valid inst.
+                // Since decode contains "continue" it's important to check at the beginning of the loop for a
+                // watchpoint trigger
+                if(triggeredWatchpoint) {
```

Comment:
Do you think there's a way to avoid this check in every iteration by using an exception? That's fired after access to the triggering instruction is done processing or so, so it's only checked for relevant instructions?

(I've yet to do a (as usual very coarse) perf comparison, so this may be nothing, just wondering.)

Reply:
I'm not sure of a good way to achieve that. I can't throw an exception the moment the watchpoint is triggered, because the full instruction needs to finish (otherwise we'd get partially done instructions and that would be a nightmare). I could check the flag at the end of every instruction that does a load or store, but there are ~55 of them and manually checking at the end of every single one seems tedious. I guess its doable though

Reply:
Right, I suspected as much, thanks for confirming.  Let's keep it as-is for now, then, that's fine.  Long term I'm thinking it might make sense to more heavily lean into code-gen for stuff like this, potentially even generate two class variants, one with and one without debugging support.